packages/controls/src/widget_string.ts

```diff
@@ -21,6 +25,105 @@ import * as _ from 'underscore';
  */
 const INVALID_VALUE_CLASS = 'jpwidgets-invalidComboValue';
 
+export
+class StringStyleModel extends DescriptionStyleModel {
+    defaults(): Backbone.ObjectHash {
+        return _.extend(super.defaults(), {
+            _model_name: 'StringStyleModel',
+            _model_module: '@jupyter-widgets/controls',
+            _model_module_version: JUPYTER_CONTROLS_VERSION,
+        });
+    }
+
+    public static styleProperties = {
+        ...DescriptionStyleModel.styleProperties,
+        background_color: {
+            selector: '',
+            attribute: 'background-color',
+            default: null as any
+        },
+        font_size: {
+            selector: '',
+            attribute: 'font-size',
+            default: ''
+        },
+        text_color: {
+            selector: '',
+            attribute: 'color',
+            default: ''
+        }
+    };
+}
+
+export
+class LabelStyleModel extends StringStyleModel {
+    defaults(): Backbone.ObjectHash {
+        return _.extend(super.defaults(), {
+            _model_name: 'LabelStyleModel',
+            _model_module: '@jupyter-widgets/controls',
+            _model_module_version: JUPYTER_CONTROLS_VERSION,
+        });
+    }
+
+    public static styleProperties = {
+        ...StringStyleModel.styleProperties,
+        font_family: {
+            selector: '',
+            attribute: 'font-family',
+            default: ''
+        },
+        font_style: {
+            selector: '',
+            attribute: 'font-style',
+            default: ''
+        },
+        font_variant: {
+            selector: '',
+            attribute: 'font-variant',
+            default: ''
+        },
+        font_weight: {
+            selector: '',
+            attribute: 'font-weight',
+            default: ''
+        },
+        text_decoration: {
+            selector: '',
+            attribute: 'text-decoration',
+            default: ''
+        }
+    };
+}
+
+export
+class TextStyleModel extends DescriptionStyleModel {
+    defaults(): Backbone.ObjectHash {
+        return _.extend(super.defaults(), {
+            _model_name: 'TextStyleModel',
+            _model_module: '@jupyter-widgets/controls',
+            _model_module_version: JUPYTER_CONTROLS_VERSION,
+        });
+    }
+
+    public static styleProperties = {
+        ...DescriptionStyleModel.styleProperties,
+        background_color: {
+            selector: '.input',
```

Comment:
I think this should just be `'input'`, not `'.input'`? In CSS, prefixing the word with a dot means to look for a CSS class `input`, whereas not having a dot means to look for a tag with name `input`.

Reply:
yes, that was meant to be `input` as a CSS class (which I have also added in the HTML display).
We could also consider the tag. If that works (I have not tested that), and if we can promise to have such an input tag for each and every text input.

Reply:
... which is not the case with `textarea.`